### PR TITLE
mkdir should not be recursive by default

### PR DESCRIPTION
**Pull request by ry**
cc @cody

### CODE REVIEW

**Comment by mactenm** [comment]
i don't know with file is is can't find it

**Comment by mactenm** [comment]
could it be debug rebut

**Comment by mactenm** [comment]
i pressed the wrong wanted to pres improve

File: js/mkdir.ts [hunk 1]
```diff
@@ -3,33 +3,48 @@ import * as msg from "gen/msg_generated";
...
+ * `mode` sets permission bits (before umask) on UNIX and does nothing on
+ * Windows.
  *
  *       import { mkdirSync } from "deno";
  *       mkdirSync("new_dir");
+ *       mkdirSync("nested/directories", true);
  */
-export function mkdirSync(path: string, mode = 0o777): void {
-  dispatch.sendSync(...req(path, mode));
+export function mkdirSync(path: string, recursive = false, mode = 0o777): void {
```
Comment (piscisaureus) [inline js/mkdir.ts]: This function should not take 3 arguments. Make the second argument an options object. So it effectively becomes:

```ts
interface MkdirOptions {
  recursive: boolean;
  mode: number;
}
`mkdirSync(path, { recursive = false, mode = 0o777 }: MkdirOptions = {})` {
...
```

File: js/mkdir.ts [hunk 2]
```diff
@@ -3,33 +3,48 @@ import * as msg from "gen/msg_generated";
...
+ * `mode` sets permission bits (before umask) on UNIX and does nothing on
+ * Windows.
  *
  *       import { mkdir } from "deno";
  *       await mkdir("new_dir");
+ *       await mkdir("nested/directories", true);
  */
-export async function mkdir(path: string, mode = 0o777): Promise<void> {
-  await dispatch.sendAsync(...req(path, mode));
+export async function mkdir(
```
Comment (piscisaureus) [inline js/mkdir.ts]: See mkdirSync comment. https://github.com/denoland/deno/pull/1530/files#r247990113

File: js/mkdir.ts (hunk 1 above)
Comment (ry) [reply js/mkdir.ts]: Done.

**Comment by ry** [comment]
Maybe

**Pull request merged**